modules/sov-ibc-transfer/src/context.rs

```diff
@@ -269,16 +261,24 @@ where
         channel_id: &ChannelId,
         coin: &PrefixedCoin,
     ) -> Result<(), TokenTransferError> {
+        let token_id = TokenId::from_str(&coin.denom.to_string()).map_err(|e| {
+            TokenTransferError::InvalidCoin {
+                coin: coin.to_string(),
+            }
+        })?;
+
+        if let Some(token_name) = self
+            .ibc_transfer
+            .minted_token_id_to_name
+            .get(&token_id, *self.working_set.borrow_mut())
+        {
+            return Err(TokenTransferError::Other(
+                    format!("Token with ID '{token_id}' is an IBC-created token and cannot be unescrowed. Use '{token_name}' as denom for receiving an IBC token from the source chain")
+                ));
+        }
```

Comment:
I feel this is redundant if the counterparty chain is executing honestly.

Reply:
But then, we also have the error case `escrowed_balance < amount` - which should never happen if the counterparty chain is running honestly.

Reply:
where I commented, is inside `unescrow_coins_validate`. 

anyway, I am okay with keeping these checks - as this doesn't add anything extra from the user's side. And, for an honest counterparty - this will be successful.

but add a comment here that there is no way to make this fail from an honest counterparty chain. because this method only fails when the counterparty chain produces a malicious IBC transfer `send_packet`.

Reply:
My mistake, the IDE was showing the comment under the escrow part.
I see your point, but yes, let's keep it. Added the comment: [4a42433](https://github.com/informalsystems/sovereign-ibc/pull/133/commits/4a424331cf9084a10d9899cde944bedf490864f2)

Reply:
no worries :smile: I too get confused among the Github and VSCode UI sometimes :sweat_smile: